src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -1,6 +1,58 @@
 @model EditUserViewModel
-@using OrchardCore.Entities
-@using OrchardCore.Users.Models
+@using OrchardCore.Entities;
+@using OrchardCore.Users.Models;
+@using Microsoft.AspNetCore.Identity;
+@using Microsoft.AspNetCore.Html;
+@inject Microsoft.Extensions.Options.IOptions<IdentityOptions> IdentityOptions
+@{
+    var options = IdentityOptions.Value;
+    var passwordOptions = new HtmlContentBuilder();
+    IHtmlContent separator = HtmlString.Empty;
+    string passwordRegex = "";
+
+    if (options.Password.RequireNonAlphanumeric)
+    {
+        passwordOptions.AppendHtml(separator).AppendHtml(T["one non-alphanumeric"]);
+        separator = new HtmlString(", ");
+        passwordRegex += @"(?=.*[^A-Za-z0-9])";
+    }
+
+    if (options.Password.RequireUppercase)
+    {
+        passwordOptions.AppendHtml(separator).AppendHtml(T["one uppercase"]);
+        separator = new HtmlString(", ");
+        passwordRegex += @"(?=.*[A-Z])";
+    }
+
+    if (options.Password.RequireLowercase)
+    {
+        passwordOptions.AppendHtml(separator).AppendHtml(T["one lowercase"]);
+        separator = T[", "];
+        passwordRegex += @"(?=.*[a-z])";
+    }
+
+    if (options.Password.RequireDigit)
+    {
+        passwordOptions.AppendHtml(separator).AppendHtml(T["one digit"]);
+        separator = new HtmlString(", ");
+        passwordRegex += @"(?=.*[0-9])";
+    }
+
+    if (separator != HtmlString.Empty)
+    {
+        separator = T[" and "];
+    }
+
+    passwordOptions.AppendHtml(separator).AppendHtml(T["{0} characters in total", options.Password.RequiredLength]);
+    passwordRegex += $".{{{options.Password.RequiredLength},}}";
+
+    if (options.Password.RequiredUniqueChars > 1)
+    {
+        passwordOptions.AppendHtml(T[", with {0} unique characters", options.Password.RequiredUniqueChars]);
+    }
+
+    var passwordTooltip = T["Password must have at least {0}.", passwordOptions];
```

Comment:
Why this logic needs to be different than ChangePassword screen and Register screen?

Reply:
IMO this should not be part of View - and Let Identity handle the messages as one could implement their own `IPasswordValidator`

Reply:
I agree with you. I tried to reproduce the code used in [OrchardCore.Setup](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Setup/Views/Setup/Index.cshtml#L18-L66) this logic should be extracted into some sort of service. also the `strength.js` resource should be registered separated into resource so it can be reused here too. I just did not want to add many changes here to keep the PR minimal

Reply:
I changed the view without the logic in the Setup module since the password change views that logic isn't the same